core/src/main/kotlin/org/evomaster/core/problem/rest/service/AIResponseClassifier.kt

```diff
@@ -11,46 +11,30 @@ import org.evomaster.core.problem.rest.classifier.GaussianOnlineClassifier
 import javax.annotation.PostConstruct
 
 
-class AIResponseClassifier : AIModel {
 
-    @Inject
-    private lateinit var config : EMConfig
-
-    private lateinit var delegate: AIModel
-
-    @PostConstruct
-    fun initModel(){
-
-        val dimension = 1 //FIXME
-
-        when(config.aiModelForResponseClassification){
-            EMConfig.AIResponseClassifierModel.GAUSSIAN -> {
-                delegate = GaussianOnlineClassifier(dimension)
-            }
-            EMConfig.AIResponseClassifierModel.GLM -> {
-                delegate = GLMOnlineClassifier(dimension, config.aiResponseClassifierLearningRate)
-            }
-            EMConfig.AIResponseClassifierModel.NN -> {
-                //TODO
-            }
-            EMConfig.AIResponseClassifierModel.NONE -> {
-                //TODO
-                delegate = object : AIModel {
-                    override fun updateModel(input: RestCallAction, output: RestCallResult) {}
-                    override fun classify(input: RestCallAction) = AIResponseClassification()
-                }
-            }
+class AIResponseClassifier(
+    private val config: EMConfig = EMConfig(),
+    private val explicitModel: AIModel? = null
+) : AIModel {
```

Comment:
the injection here shouldn't be touched.  constructor should have no inputs. long story...
recall this class is instantiated as singleton with a DI framework